shell.nix

```diff
@@ -1,19 +1,43 @@
 scope@{ pkgs ? import <nixpkgs> { } }:
+with pkgs;
+let
+  tokio-console = import ./default.nix { inherit pkgs; };
 
-let env = (import ./default.nix scope);
+  haveGlibcLocales = pkgs.glibcLocales != null && stdenv.hostPlatform.libc == "glibc";
+  glibcLocales-utf8 = (glibcLocales.override { locales = [ "en_US.UTF-8" ]; });
 
-in with pkgs;
+  env = buildEnv {
+    name = "console-env";
+    paths = [
+      direnv
+      binutils
+      stdenv
+      bashInteractive
+      docker
+      cacert
+      gcc
+      cmake
+      pkg-config
+      openssl
+    ]
+    ++ lib.optional haveGlibcLocales glibcLocales-utf8
+    ++ lib.optionals stdenv.isDarwin [ libiconv ]
+    ++ tokio-console.buildInputs
+    ++ tokio-console.nativeBuildInputs;
+  };
+in
 mkShell {
+  buildInputs = [ env ];
+
   PROTOC = "${protobuf}/bin/protoc";
   PROTOC_INCLUDE = "${protobuf}/include";
-  LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive";
+  LOCALE_ARCHIVE = lib.optionalString haveGlibcLocales "${glibcLocales-utf8}/lib/locale/locale-archive";
   LC_ALL = "en_US.UTF-8";
   OPENSSL_DIR = "${openssl.dev}";
   OPENSSL_LIB_DIR = "${openssl.out}/lib";
   SSL_CERT_FILE = "${cacert}/etc/ssl/certs/ca-bundle.crt";
   GIT_SSL_CAINFO = "${cacert}/etc/ssl/certs/ca-bundle.crt";
   CURL_CA_BUNDLE = "${cacert}/etc/ca-bundle.crt";
   CARGO_TERM_COLOR = "always";
-  RUST_BACKTRACE = "1";
-  buildInputs = [ (import ./default.nix { inherit pkgs; }) ];
+  RUST_BACKTRACE = "full";
```

Comment:
Aaaand one final "not sure if needed." The package builds fine without all of these variables set. Maybe just the `RUST_BACKTRACE` and `CARGO_TERM_COLOR`?

Reply:
i'm fine with removing anything that isn't necessary to build the package --- again, i copied and pasted the config from a different nix derivation originally, and forgot to remove some stuff!

i think the `PROTOC` and `PROTOC_INCLUDE` env vars are probably necessary, but I think we can safely remove all the openssl stuff...